distributed/worker.py

```diff
@@ -2953,6 +2966,8 @@ def release_key(
             if key in self.threads:
                 del self.threads[key]
 
+            # task is released but still running - set thread exception
+            # whether or not to do this automatically should be configurable
```

Comment:
This comment feels a bit out of place. Should this be part of `transition_executing_released`?